src/_zkapauthorizer/model.py

```diff
@@ -278,11 +278,14 @@ def from_node_config(cls, node_config, now, connect=None):
             conn,
         )
 
-    @with_cursor
-    def call_if_empty(self, cursor, f: Callable[[Cursor], None]) -> None:
+    @with_cursor_async
+    def call_if_empty(self, cursor, f: Callable[[Cursor], _T]) -> _T:
         """
         Transactionally determine that the database is empty and call the given
         function if it is or raise ``NotEmpty`` if it is not.
+
+        The function may return an ``Awaitable``.  If it does the transaction
+        opened for it will be kept open until the ``Awaitable`` completes.
```

Comment:
I haven't done much "typing in python" but the function signature doesn't say this ... can it?
(Or I guess is there a "async-def Callable" type-signature?)

Reply:
Oops.  The "may" here is left-over from when I tried to support both synchronous and asynchronous functions here but I gave up on that.  Now it *must* be an async function.

As far as an "async-def Callable", as far as I know this is written as `Callable[..., Awaitable[...]]`.

Since `f` _must_ be an async function now, the `_T` in the signature should probably be replaced by `Awaitable[_T]`, making it:

```py
def call_if_empty(self, cursor, f: Callable[[Cursor], Awaitable[_T]]) -> Awaitable[_T]:
```

So - `f` is a function that returns an `Awaitable` that completes with a `_T` and `call_if_empty` is a function that returns the same.

How does that look?